File: código-fonte-reforma/src/main/java/com/caldeira/service/XmlItem.java
package com.caldeira.service;

/** Item (grupo det) de uma NF-e. NFS-e não tem itens — ver {@link XmlNota}. */
class XmlItem {

    int numItem;
    String codProduto;
    String descricao;
    String ncm;
    String exTipi;
    String cfop;
    String unidadeComercial;
    String unidadeTributavel;
    double qtdComercial;
    double qtdTributavel;
    double vlUnitario;
    /** vProd — valor autoritativo do item; nunca recalculado a partir de qtd x unitário. */
    double vlProduto;
    /** vItem quando presente (vProd + IPI etc.); cai para vlProduto quando ausente. */
    double vlContabil;
    double vlDesconto;
    double vlFrete;
    double vlSeguro;
    double vlOutras;

    // ICMS
    String cstIcms;
    String origemIcms;
    double baseIcms;
    double aliqIcms;
    double vlIcms;

    // IPI
    String cstIpi;
    String codEnquadramentoIpi;
    double baseIpi;
    double aliqIpi;
    double vlIpi;

    // Imposto de importação (grupo imposto/II — só existe em nota de importação)
    double baseIi;
    double vlIi;
    double vlDespesasAduaneiras;
    double vlIof;

    // PIS / COFINS
    String cstPis;
    double basePis;
    double aliqPis;
    double vlPis;
    String cstCofins;
    double baseCofins;
    double aliqCofins;
    double vlCofins;

    /** Grupo det/prod/DI — presente apenas em notas de importação (CFOP 3xxx). */
    XmlDi di;

    /** Grupo IBSCBS do item (Reforma Tributária) — lido, não gravado nos SAFX 04/07/08/49. */
    XmlIbsCbs ibsCbs;
}

File: ABMC.RANFE.LIGHT_SQL/ABMC.RANFE.LIGHT_SQL/ReformaTributaria/Modelo/XmlItem.cs
namespace ABMC.RANFE.LIGHT_SQL.ReformaTributaria.Modelo
{
    /// <summary>Item (grupo det) de uma NF-e. NFS-e não tem itens — ver <see cref="XmlNota"/>.</summary>
    internal sealed class XmlItem
    {
        internal int NumItem;
        internal string CodProduto;
        internal string Descricao;
        internal string Ncm;
        internal string ExTipi;
        internal string Cfop;
        internal string UnidadeComercial;
        internal string UnidadeTributavel;
        internal double QtdComercial;
        internal double QtdTributavel;
        internal double VlUnitario;

        /// <summary>vProd — valor autoritativo do item; nunca recalculado a partir de qtd x unitário.</summary>
        internal double VlProduto;

        /// <summary>vItem quando presente (vProd + IPI etc.); cai para VlProduto quando ausente.</summary>
        internal double VlContabil;
        internal double VlDesconto;
        internal double VlFrete;
        internal double VlSeguro;
        internal double VlOutras;

        // ICMS
        internal string CstIcms;
        internal string OrigemIcms;
        internal double BaseIcms;
        internal double AliqIcms;
        internal double VlIcms;

        // IPI
        internal string CstIpi;
        internal string CodEnquadramentoIpi;
        internal double BaseIpi;
        internal double AliqIpi;
        internal double VlIpi;

        // Imposto de importação (grupo imposto/II — só existe em nota de importação)
        internal double BaseIi;
        internal double VlIi;
        internal double VlDespesasAduaneiras;
        internal double VlIof;

        // PIS / COFINS
        internal string CstPis;
        internal double BasePis;
        internal double AliqPis;
        internal double VlPis;
        internal string CstCofins;
        internal double BaseCofins;
        internal double AliqCofins;
        internal double VlCofins;

        /// <summary>Grupo det/prod/DI — presente apenas em notas de importação (CFOP 3xxx).</summary>
        internal XmlDi Di;

        /// <summary>Grupo IBSCBS do item (Reforma Tributária).</summary>
        internal XmlIbsCbs IbsCbs;
    }
}
